packages/app/src/components/Auth/AuthProvider.tsx

```diff
@@ -31,24 +32,36 @@ const AuthContext = React.createContext<AuthContextValue>({
   },
 });
 
+const LOCAL_STORAGE_CLIENT_KEY = "idp-auth-client-id";
+
 export const AuthProvider = ({ children }: AuthProviderProps) => {
   const [accessToken, setAccessToken] = useState<AccessToken>();
   const [userInfo, setUserInfo] = useState<UserInfo>();
 
   const { auth } = useConfig();
 
+  const [clientId, setClientId] = React.useState(
+    localStorage.getItem(LOCAL_STORAGE_CLIENT_KEY)
+  );
   useEffect(() => {
+    localStorage.setItem(LOCAL_STORAGE_CLIENT_KEY, clientId ?? "");
+  }, [clientId]);
+
+  useEffect(() => {
+    if (!clientId) {
+      return;
+    }
     const initAuth = async () => {
       if (!auth) {
         return;
       }
       if (!AuthClient.client) {
-        if (!auth.clientId) {
+        if (!clientId) {
           throw new Error(
             "Please add a valid client ID in the .env.local file and restart the application"
           );
```

Comment:
This should probably just return if there is no clientId as we are now taking it from the localstorage instead of the .env.local file, and it will be handled in the UI. (And having no client Id is a normal workflow now, not an error.
```suggestion
        if (!clientId) {
          return;
```